src/controllers/product.controller.js

```diff
@@ -37,3 +44,29 @@ export const getAllProducts = async (req, res) => {
       .json({ error: err.message, message: 'Failed to retrieve products' });
   }
 };
+export const getSpecificProduct = async (req, res) => {
+  try {
+    const { productId } = req.params;
+    const { role, id } = req.user;
+    const product = await ProductService.getSingleProduct(productId);
+    if (product) {
+      if (role === 'seller') {
+        if (product.userId != id)
+          return res
+            .status(404)
+            .json({ message: "Product doesn't exist in your collection" });
+      }
+      if (role === 'buyer') {
+        if (!product.available)
+          return res.status(404).json({ message: 'Product is not available' });
+      }
```

Comment:
use middleware to check this

Reply:
In addition, the middleware should check if the product exists